### PR TITLE
✨ 374 New File Names

### PR DESCRIPTION
**Pull request by xoscar**
This PR is the first PR based on the documentation https://kubeshop.atlassian.net/wiki/spaces/KUBESHOP/pages/486309895/Post+MVP+Frontend+Improvements that updates the front end architecture to be more flexible and scalable.

## Changes

- Introduces a new pattern for file names depending on the type.
- Divides resource spaces into entities and their own folder.
- Adds the gateways folder and pattern for I/O.
- Introduces a standard way of adding types.

## Fixes

- Duplicated types.
- Unused types.

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

File: web/src/components/Navigation/index.tsx [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// eslint-disable-next-line no-restricted-exports
+export {default} from './Router';
```
Comment (xoscar) [inline web/src/components/Navigation/index.tsx:2]: Moving the Router under the components folder

File: web/src/components/Trace/Trace.tsx [hunk 2]
```diff
@@ -10,30 +10,32 @@ import {CloseCircleFilled} from '@ant-design/icons';
...
 
-import {AssertionResult, ISpan, TestRunResult, TestState} from 'types';
-
 import {
   useGetTestByIdQuery,
   useGetTestResultByIdQuery,
   useRunTestMutation,
   useUpdateTestResultMutation,
-} from 'redux/services/TestService';
+} from 'gateways/Test.gateway';
```
Comment (xoscar) [inline web/src/components/Trace/Trace.tsx:18]: This are now coming from the gateway file

File: web/src/entities/Assertion/Assertion.service.ts [hunk 3]
```diff
@@ -1,34 +1,11 @@
...
-  ItemSelector,
-  ITrace,
-  LOCATION_NAME,
-  ResourceSpan,
-  SpanAssertionResult,
-} from 'types';
-import {escapeString} from '../utils';
-import {getSpanValue} from './SpanService';
-
-const getOperator = (op: COMPARE_OPERATOR) => {
```
Comment (xoscar) [inline web/src/entities/Assertion/Assertion.service.ts:16]: Moving this to the Operator Service

File: web/src/entities/GuidedTour/GuidedTour.service.ts [hunk 4]
```diff
@@ -11,7 +11,7 @@ type TTour = Record<GuidedTours, boolean>;
 
 const GUIDED_TOUR_KEY = 'guided_tour';
 
-const {get, set} = LocalStorageService<TTour>(GUIDED_TOUR_KEY);
+const {get, set} = LocalStorageGateway<TTour>(GUIDED_TOUR_KEY);
```
Comment (xoscar) [inline web/src/entities/GuidedTour/GuidedTour.service.ts:14]: Gateways  = I/O
Services = Business logic

File: web/src/entities/Operator/Operator.service.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import {CompareOperator, CompareOperatorNameMap, CompareOperatorSymbolMap} from './Operator.constants';
```
Comment (xoscar) [inline web/src/entities/Operator/Operator.service.ts:1]: Moving forward services will be static functions that provide an object with methods and attributes. This allows the possibility of composing if required.

File: web/src/entities/GuidedTour/GuidedTour.service.ts (hunk 4 above)
Comment (schoren) [reply web/src/entities/GuidedTour/GuidedTour.service.ts:14]: nice

**Pull request merged**